dev/Interop/StoragePickers/FileTypeChoicesMap.cpp

```diff
@@ -30,42 +53,160 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
+        }
+        else
+        {
+            // Key doesn't exist, insert new pair at the end to maintain insertion order
+            size_t newIndex = m_orderedMap.size();
+            m_orderedMap.emplace_back(key, validatingVector);
+            m_keyToIndex[std::wstring(key.c_str())] = newIndex;
+            return true; // Indicates new insertion
+        }
     }
 
     winrt::Windows::Foundation::Collections::IVector<hstring> FileTypeChoicesMap::Lookup(hstring const& key) const
     {
-        return m_innerMap.Lookup(key);
+        auto it = FindKey(key);
+        if (it != m_orderedMap.end())
+        {
+            return it->second;
+        }
+        throw winrt::hresult_out_of_bounds(L"Key not found");
     }
 
     uint32_t FileTypeChoicesMap::Size() const
     {
-        return m_innerMap.Size();
+        return static_cast<uint32_t>(m_orderedMap.size());
     }
 
     bool FileTypeChoicesMap::HasKey(hstring const& key) const
     {
-        return m_innerMap.HasKey(key);
+        return FindKey(key) != m_orderedMap.end();
     }
 
     winrt::Windows::Foundation::Collections::IMapView<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>> FileTypeChoicesMap::GetView() const
     {
-        return m_innerMap.GetView();
+        return make<OrderedMapView>(m_orderedMap);
     }
 
     void FileTypeChoicesMap::Remove(hstring const& key)
     {
-        m_innerMap.Remove(key);
+        size_t index = FindKeyIndex(key);
+        if (index != SIZE_MAX)
+        {
+            // Remove from vector
+            m_orderedMap.erase(m_orderedMap.begin() + index);
+            
+            // Remove from hash map
+            m_keyToIndex.erase(std::wstring(key.c_str()));
+            
+            // Update indices for all elements after the removed one
+            for (auto& kvp : m_keyToIndex)
+            {
+                if (kvp.second > index)
+                {
+                    kvp.second--;
+                }
+            }
+        }
     }
 
     void FileTypeChoicesMap::Clear()
     {
-        m_innerMap.Clear();
+        m_orderedMap.clear();
+        m_keyToIndex.clear();
     }
 
     winrt::Windows::Foundation::Collections::IIterator<winrt::Windows::Foundation::Collections::IKeyValuePair<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>>> FileTypeChoicesMap::First() const
     {
-        return m_innerMap.First();
+        return make<OrderedMapIterator>(m_orderedMap);
+    }
+
+    // OrderedMapIterator implementation
+    winrt::Windows::Foundation::Collections::IKeyValuePair<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>> OrderedMapIterator::Current() const
+    {
+        if (m_current >= m_map.size())
+        {
+            throw winrt::hresult_out_of_bounds();
+        }
+        // Create a simple key-value pair
+        auto tempMap = single_threaded_map<hstring, winrt::Windows::Foundation::Collections::IVector<hstring>>();
```

Comment:
NIT: `auto tempMap{ ... };`

[ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es23-prefer-the--initializer-syntax)

same comment applies throughout